doc/tutorials/inter-vm_communication.rst

```diff
@@ -0,0 +1,486 @@
+
+Inter-VM vUART
+**************
+
+Inter-VM vUART communication is based on the vUART implementation. It is
+used to transfer data between two VMs at low speed. (Refer to :ref:`vuart_virtualization`
+and :ref:`vuart_config`).
+
+:numref:`Inter-VM vUART communication` shows the Inter-VM vUART communication overview:
+
+.. figure:: images/Inter-VM_vUART_communocation_overview.png
+   :align: center
+   :name: Inter-VM vUART communication
+
+   Inter-VM vUART communication
+
+- Pros:
+   - POSIX APIs.
+   - UART driver is available in most VM, development-friendly.
+
+- Cons:
+   - The communication rate is low.
+   - Need to configure communication vUART in hypervisor scenario.
+
+Inter-VM network communication
+******************************
+
+Inter-VM network communication is based on the network stack. ACRN supports
+both pass-through NICs to VM and Virtio-Net solutions. (Refer to :ref:`virtio-net`
+background introductions of ACRN Virtio-Net Architecture and Design). 
+
+:numref:`Inter-VM network communication` shows the Inter-VM network communication overview:
+
+.. figure:: images/Inter-VM_network_communication_overview.png
+   :align: center
+   :name: Inter-VM network communication
+
+   Inter-VM network communication
+
+- Pros:
+   - Socket-based APIs; development friendly.
+
+- Cons:
+   - Multiple layers are involved across the data path.
+   - Potential more CPU overhead in SOS if using Virtio-net.
+
+Inter-VM shared memory communication (ivshmem)
+**********************************************
+
+Inter-VM shared memory communication is based on a shared memory mechanism
+to transfer the data between VMs. The ACRN device model or hypervisor emulates
+a virtual PCI device (called an ivshmem device) to expose this shared memory's
+base address and size. (Refer to :ref:`ivshmem-hld` and :ref:`enable_ivshmem` for the
+background introductions).
+
+:numref:`Inter-VM shared memory communication` shows the Inter-VM shared memory communication overview:
+
+.. figure:: images/Inter-VM_shared_memory_communication_overview.png
+   :align: center
+   :name: Inter-VM shared memory communication
+
+   Inter-VM shared memory communication
+
+- Pros:
+   - Shared memory is exposed to VMs via PCI MMIO Bar and is mapped and accessed directly.
+   - The data path is short.
+   - No SOS emulation is involved if using hv-land ivshmem.
+
+- Cons:
+   - Applications talk to the device directly (via UIO on Linux and Ivshmem driver on WAAG).
+   - Applications need to implement protocols such as a handshake, data transfer, and data
+     integrity.
+
+
+How to implement an Ivshmem application on ACRN
+***********************************************
+
+As Ivshmem is a PCI device that shares memory inside a virtual machine, the application
+running on the VM can access it as a standard PCI device. Besides that, the application
+needs to design a data transfer notification mechanism between the VMs.
+
+- Initial Device:
+
+  Enable and bind to UIO PCI driver
```

Comment:
Please turn the sequence below into a `code-block`

Reply:
fixed.

Reply:
Thanks, `sudo` has been added.